### ISSUE TITLE
Odd behavior for disabled in arrays (setting model)

### ISSUE THREAD

**Issue by FritzHerbers** [ngx-formly/ngx-formly#3160]
Since 5.10.20 there is a regression in how disabled behaves for arrays.

There are different scenarios, a description you find in the file.

https://stackblitz.com/edit/angular-mjbvnf-azpgeu?file=src%2Fapp%2Fapp.component.ts
Also question:
At fields load, f1->disabled=true, s3->disabled=false gets disabled, is this wrong or as intended ?

https://stackblitz.com/edit/angular-mjbvnf-ge9ysl?file=src%2Fapp%2Fapp.component.ts

I had a similar problem for records, which was fixed: 
https://github.com/ngx-formly/ngx-formly/issues/1743
https://github.com/ngx-formly/ngx-formly/issues/2907

- Angular version: 12.2.16
- Formly version: 5.10.31

**Comment by FritzHerbers** [comment]
@aitboudad 
Thanks for the fix.

Is the following as expected, a child disabled setting is ignored when field disabled= true ?
And, at a model change, the field disabled=true is again propagated to the children ?

```
  it('should keep disabled state of field group consistent after detect changes - 3', () => {
    app = {
      fields: [
        {
          key: 'f1',
          templateOptions: { disabled: true },
          fieldGroup: [
            { key: 's1', templateOptions: { disabled: true } },
            { key: 's2', templateOptions: { disabled: false } },  // this setting is ignored
          ],
        },
      ],
      form: new FormGroup({}),
      options: {},
      model: {},
    };

    const fixture = createTestComponent('<formly-form [form]="form" [fields]="fields" [model]="model" [options]="options"></formly-form>');

    let field = fixture.componentInstance.fields[0];
    expect(field.templateOptions.disabled).toEqual(true);
    expect(field.fieldGroup[0].templateOptions.disabled).toEqual(true);
    expect(field.fieldGroup[1].templateOptions.disabled).toEqual(true); // ignored setting

    field.fieldGroup[1].templateOptions.disabled = false;
    expect(field.fieldGroup[1].templateOptions.disabled).toEqual(false); // correct

    fixture.componentInstance.model = {};
    fixture.detectChanges();

    field = fixture.componentInstance.fields[0];
    expect(field.templateOptions.disabled).toEqual(true);
    expect(field.fieldGroup[0].templateOptions.disabled).toEqual(true);
    expect(field.fieldGroup[1].templateOptions.disabled).toEqual(true); // ignored previous set disabled=false
  });
```

**Comment by aitboudad** [comment]
> Is the following as expected, a child disabled setting is ignored when field disabled= true ?

That's expected, when parent field (with key) is disabled, child must be disabled as well.

**Comment by FritzHerbers** [comment]
@aitboudad 
Thanks for your reply.
Looking forward for an updated release.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.32](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.32) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
